server/Makefile

```diff
@@ -213,6 +213,7 @@ include build/*.mk
 include public/Makefile
 
 LDFLAGS += -X "github.com/mattermost/mattermost/server/public/model.MockCWS=$(MM_ENABLE_CWS_MOCK)"
+LDFLAGS += -X "github.com/mattermost/mattermost/server/public/model.GiphySdkKey=$(MM_GIPHY_SDK_KEY)"
```

Comment:
Will this be set only for prod? We want to make sure that for test and dev we have the default test key below

Reply:
i was able to get it working by explicitly passing the flag as
`make XXX MM_GIPHY_SDK_KEY=helooo`

and for dev, passing empty the flag takes the default from `config.mk`

Let me know if this is not the right approach

Reply:
In this case we would need to set the env var in the test and prod builds differently? 

Wondering if we can keep the service environment check we had before just for test and always set the key in the build but it will override if you are working on a test and/or dev environment. This way we can minimize the changes in the pipeline

Reply:
i added back the service environment, and defaulted for production the flag, is this how you suggested?